metrics/heapster.go

```diff
@@ -213,12 +212,12 @@ func getListersOrDie(kubernetesUrl *url.URL) (*cache.StoreToPodLister, *cache.St
 	return podLister, nodeLister
 }
 
-func createKubeClientOrDie(kubernetesUrl *url.URL) *kube_client.Client {
+func createKubeClientOrDie(kubernetesUrl *url.URL) *kube_client.Clientset {
```

Comment:
it's a legacy thing -- you run `--source=kubernetes.summary_api=$KUBE_CONFIGURATION_IN_A_URL`, but you can specify to use the inClusterConfig option which should just pull from the injected kubeconfig, IIRC.